### ISSUE TITLE
Cannot edit math block contents from the keyboard

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#73446]
### Description

In the editor, there is no path to edit the expression in a math block using only the keyboard.

### Step-by-step reproduction instructions

1. Create a math block. Add a simple expression.
2. Place the cursor in a block before or after the block.
3. Use the arrow keys to navigate between blocks. Attempt to move the cursor into the math editor.

I was unable to find a way to do this. I could select the expression, but focus was not moved into the textarea to edit the expression.

### Environment info

- WP 6.9-RC2-61266, Gutenberg 22.1.0
- Chrome & Firefox, Windows 11

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
